website/ref/language.md

```diff
@@ -2358,10 +2359,16 @@ The following namespaces have special meanings to the language:
     an external command.
 
 -   `E:` refers to environment variables. For instance, `$E:USER` is the
-    environment variable `USER`.
-
-    This **is** always needed, because unlike command resolution, variable
-    resolution does not fall back onto environment variables.
+    environment variable `USER`. If the environment variable does not exist it
+    expands to an empty string. If you care that the env var is not set use
+    [`has-env`](./builtin.html#has-env) or [`get-env`](./builtin.html#get-env).
+    This behavior makes it easy to deal with the common case where you don't
+    care if the env var is unset, or set to an empty string, by doing this:
+    `eq $E:UNSET_VAR ''`.
```

Comment:
> If you care that the env var is not set ...
>
> ... where you don't care if the env var is set

This is a bit hard to read, and it isn't quite clear what "caring" exactly entails. This would be more explicit and less ambiguous:

> If you want to distinguish between environment variables that are unset and those that are set but empty ...
> 
> ... where you don't need to distinguish unset and empty environment variables